preprocessing/shift_generation.py

```diff
@@ -400,9 +406,13 @@ def get_t_covered_by_shift(shifts, time_sets):
     combined_time_periods = time_sets["combined_time_periods"][0]
     t_covered_by_shift = {}
     for shift in shifts:
-        end = combined_time_periods.index(shift[0] + shift[1] - time_step)
-        start = combined_time_periods.index(shift[0])
-        t_covered_by_shift[shift[0], shift[1]] = combined_time_periods[start : (end + 1)]
+        try:
+            end = combined_time_periods.index(shift[0] + shift[1] - time_step)
+            start = combined_time_periods.index(shift[0])
+            t_covered_by_shift[shift[0], shift[1]] = combined_time_periods[start: (end + 1)]
+        except:
+            pass
+
```

Comment:
The try-block is implemented since predefined shift not corresponding to daily demand (and that thus have start_time and end_time with no demand) may occur.